### PR TITLE
docs: Add example and tests for `pl.concat()` with `Expr` input

### LINKED ISSUE #19813: Should `pl.concat()` accept `Expr` input?
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

df = pl.DataFrame({"x": [1]})

pl.concat([df, df], how = "foo")
# ValueError: DataFrame `how` must be one of {'vertical', 'vertical_relaxed', 'diagonal', 'diagonal_relaxed', 'horizontal', 'align'}, got 'foo'

df.select(pl.concat([pl.col("x"), pl.col("x")], how = "foo"))
# shape: (2, 1)
# ┌─────┐
# │ x   │
# │ --- │
# │ i64 │
# ╞═════╡
# │ 1   │
# │ 1   │
# └─────┘
```

### Log output

_No response_

### Issue description

Might simply be a documentation issue, but it's unclear to me whether `Expr` should be accepted input in`pl.concat()`. The [docs of `items`](https://docs.pola.rs/api/python/dev/reference/api/polars.concat.html) say:

> DataFrames, LazyFrames, or Series to concatenate.

They also don't mention the fact that when the inputs are `Expr`, `how` can take any value, even `"foo"` as in the example above.

### Expected behavior

Either an update in the docs saying that `Expr` are accepted and that in this case `how` doesn't matter, or forbidding `Expr` as input.

### Installed versions

<details>

```
--------Version info---------
Polars:              1.13.1
Index type:          UInt32
Platform:            Linux-6.8.0-48-generic-x86_64-with-glibc2.39
Python:              3.12.3 (main, Sep 11 2024, 14:17:37) [GCC 13.2.0]
LTS CPU:             False

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               <not installed>
cloudpickle          <not installed>
connectorx           <not installed>
deltalake            <not installed>
fastexcel            <not installed>
fsspec               <not installed>
gevent               <not installed>
great_tables         <not installed>
matplotlib           <not installed>
nest_asyncio         <not installed>
numpy                2.1.2
openpyxl             <not installed>
pandas               <not installed>
pyarrow              <not installed>
pydantic             <not installed>
pyiceberg            <not installed>
sqlalchemy           <not installed>
torch                <not installed>
xlsx2csv             <not installed>
xlsxwriter           <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by etiennebacher**
Fixes #19813

### CODE REVIEW

File: py-polars/polars/functions/eager.py [hunk 1]
```diff
@@ -31,14 +31,14 @@ def concat(
...
+    Combine multiple DataFrames, LazyFrames, Series, or Expr into a single object.
 
     Parameters
     ----------
     items
-        DataFrames, LazyFrames, or Series to concatenate.
+        DataFrames, LazyFrames, Series, or Expr to concatenate.
     how : {'vertical', 'vertical_relaxed', 'diagonal', 'diagonal_relaxed', 'horizontal', 'align'}
-        Series only support the `vertical` strategy.
+        This is not used when items are Expr. Series only support the `vertical` strategy.
```
Comment (eitsupi) [inline py-polars/polars/functions/eager.py:41]: I think `vertical` should be checked for Expr as well.

Comment (ritchie46) [reply py-polars/polars/functions/eager.py:41]:
```suggestion
        Series and Expr only support the `vertical` strategy.
```

**Comment by etiennebacher** [comment]
Test failures only appear with the new streaming engine but I have no idea why. Should I remove the new test?

**Review by ritchie46 — APPROVED** [review]

**Comment by ritchie46** [comment]
No, we should fix that.

**Comment by etiennebacher** [comment]
https://github.com/pola-rs/polars/issues/19877

**Comment by etiennebacher** [comment]
Thanks, it passes now

**Comment by eitsupi** [comment]
If you're new to Git, I recommend that you don't try to rebase.

**Comment by etiennebacher** [comment]
Not new, but I never use that (and when I do I rely on the github interface). I see it's the norm here, which is why I tried

**Comment by eitsupi** [comment]
There would have been no need to close the PR if the branch had been reset to the state it was in before the rebase attempt was made.

**Pull request closed without merging**